src/data/finsupp.lean

```diff
@@ -94,17 +115,22 @@ lemma support_subset_iff {s : set α} {f : α →₀ β} :
 by simp only [set.subset_def, mem_coe, mem_support_iff];
    exact forall_congr (assume a, @not_imp_comm _ _ (classical.dec _) (classical.dec _))
 
+/-- Given `fintype α`, `equiv_fun_on_fintype` is the `equiv` between `α →₀ β` and `α → β`.
+  (All functions on a finite type are finitely-supported.) -/
 def equiv_fun_on_fintype [fintype α] : (α →₀ β) ≃ (α → β) :=
-⟨λf a, f a, λf, mk (finset.univ.filter $ λa, f a ≠ 0) f (by simp),
+⟨λf a, f a, λf, mk (finset.univ.filter $ λa, f a ≠ 0) f (by simp only [true_and, finset.mem_univ,
+  iff_self, finset.mem_filter, finset.filter_congr_decidable, forall_true_iff]),
   begin intro f, ext a, refl end,
   begin intro f, ext a, refl end⟩
 
 end basic
 
+/-! ### Declarations about `single` -/
+
 section single
 variables [has_zero β] {a a' : α} {b : β}
 
-/-- `single a b` is the finitely supported function which has
+/-- `single a b` is the finitely-supported function which has
```

Comment:
```suggestion
/-- `single a b` is the finitely supported function which has
```
I don't think the `-` is needed here. Compare the [Google results](https://www.google.com/search?rlz=1C1CHBF_enNL860NL860&biw=1536&bih=754&sxsrf=ALeKk02NVPEbip3OaGZYb96DBWYp4xCjLA%3A1582708731125&ei=-zdWXqiiB47xkwWg9aqYAg&q=%22finitely-supported%22&oq=%22finitely-supported%22&gs_l=psy-ab.3..0i30l5.25881.26129..26418...0.0..0.73.325.5......0....1..gws-wiz.ppkfPW0epwA&ved=0ahUKEwjoi8698e7nAhWO-KQKHaC6CiM4ChDh1QMICw&uact=5)

Reply:
You're right. The rule I remembered was that compound adjectives get a hyphen, but apparently there's an exception for adverb-adjective combinations. https://en.wikipedia.org/wiki/Hyphen#Compound_modifiers